### PR TITLE
Suisin/chore: update error message for PhoneNumberTaken

### PR DESCRIPTION
**Pull request by suisin-deriv**
## Changes:

Account settings changes related to phone number verification project

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/account/src/Styles/account.scss [hunk 1]
```diff
@@ -451,6 +451,12 @@ $MIN_HEIGHT_FLOATING: calc(
                 }
             }
 
+            .dc-form-submit-error-message {
+                &__phone-number-taken {
+                    color: $color-red-1;
```
Comment (amina-deriv) [inline packages/account/src/Styles/account.scss:456]: cant we pass  `color='loss-danger'` prop for the Text component?

File: packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
+            default:
+                <Text as='p' size='xxs' weight='bold' color='prominent'>
+                    {message}
+                </Text>;
+        }
+    };
+
+    return (
+        <div className={classNames('dc-form-submit-error-message', className)}>
+            <Icon icon='IcAlertDanger' data_testid='form_submit_error' />
```
Comment (Dhruv-deriv) [inline packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx:34]: can we use Quill icon here?

File: packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
-    <div className={classNames('dc-form-submit-error-message', className)}>
-        <Icon icon='IcAlertDanger' data_testid='form_submit_error' />
-        <Text as='p' size='xxs' weight='bold' color='prominent'>
-            {message}
-        </Text>
-    </div>
-);
+const FormSubmitErrorMessage = ({ className, message, error_code }: TFormSubmitErrorMessage) => {
+    const getErrorMessage = () => {
+        switch (error_code) {
```
Comment (amina-deriv) [inline packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx:15]: switch case is only handling a single specific error code. i think simple if else case would be fine

File: packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx (hunk 2 above)
Comment (suisin-deriv) [reply packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx:34]: @Dhruv-deriv this is legacy design 🙏 there's no @deriv-com/quill-ui install in core component package.

File: packages/account/src/Sections/Profile/PersonalDetails/verify-button.tsx [hunk 4]
```diff
@@ -8,94 +8,105 @@ import { useTranslations, Localize } from '@deriv-com/translations';
...
-    const { sendPhoneNumberVerifyEmail, WS, is_loading } = useVerifyEmail('phone_number_verification');
-    const { isDesktop } = useDevice();
-    const ref = useRef<HTMLDivElement | null>(null);
+export const VerifyButton = observer(
+    ({ is_verify_button_disabled, next_email_otp_request_timer, setStatus }: TVerifyButton) => {
+        const { client, ui } = useStore();
+        const { setShouldShowPhoneNumberOTP, is_scroll_to_verify_button, setIsScrollToVerifyButton } = ui;
+        const { account_settings, setVerificationCode } = client;
+        const { phone_number_verification } = account_settings;
+        const is_phone_number_verified = !!phone_number_verification?.verified;
```
Comment (amina-deriv) [inline packages/account/src/Sections/Profile/PersonalDetails/verify-button.tsx:28]: @suisin-deriv just noticed that we are checking this condition in multiple places. should we create a hook `useIsphoneNumberVerified` instead?

File: packages/account/src/Sections/Profile/PersonalDetails/verify-button.tsx [hunk 5]
```diff
@@ -8,94 +8,105 @@ import { useTranslations, Localize } from '@deriv-com/translations';
...
-                ? (resendCodeTimer = `${localize(' in ')}${next_email_otp_request_timer}s`)
-                : (resendCodeTimer = `${localize(' in ')}${Math.round(next_email_otp_request_timer / 60)}m`);
-        } else {
-            resendCodeTimer = '';
-        }
+        const verifyTimer = () => {
+            let resendCodeTimer = '';
+            if (next_email_otp_request_timer) {
+                next_email_otp_request_timer < 60
+                    ? (resendCodeTimer = `${localize(' in ')}${next_email_otp_request_timer}s`)
```
Comment (amina-deriv) [inline packages/account/src/Sections/Profile/PersonalDetails/verify-button.tsx:73]: cab we double check if localize is working properly here?

File: packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx (hunk 2 above)
Comment (likhith-deriv) [reply packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx:34]: Hi @suisin-deriv asked a question in DM :)

File: packages/account/src/Sections/Profile/PersonalDetails/verify-button.tsx (hunk 5 above)
Comment (suisin-deriv) [reply packages/account/src/Sections/Profile/PersonalDetails/verify-button.tsx:68]: Yeappp localize is working properly here 🙏

File: packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx (hunk 2 above)
Comment (Dhruv-deriv) [reply packages/components/src/components/form-submit-error-message/form-submit-error-message.tsx:20]: LGTM @suisin-deriv 👍

**Review by Dhruv-deriv — APPROVED** [review]

**Review by likhith-deriv — APPROVED** [review]

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**